flink-download-jar-runtime/src/main/java/basic/application/StreamingJob.java

```diff
@@ -0,0 +1,262 @@
+		Properties flinkProperties = null;
+
+
+		String s3SinkPath = parameter.get("s3SinkPath", "");
+		String region = parameter.get("region", "us-east-1");
+		String inputStreamName = parameter.get("inputStreamName", "");
+		String userJarBucket = parameter.get("userJarBucket", "{AWS BUCKET NAME}");
+		String userJarKey = parameter.get("userJarKey", "{USER JAR KEY PREFIX (i.e. prefix/myfile.jar)}");
+
+		if (applicationProperties != null) {
+			flinkProperties = applicationProperties.get("FlinkApplicationProperties");
+		}
+
+		if (flinkProperties != null) {
+			s3SinkPath = flinkProperties.get("s3SinkPath").toString();
+			region = flinkProperties.get("region").toString();
+			inputStreamName = flinkProperties.get("inputStreamName").toString();
+			userJarBucket = flinkProperties.get("userJarBucket").toString();
+			userJarKey = flinkProperties.get("userJarKey").toString();
+
+		}
+
+		final String userJarFileName = userJarKey.substring(userJarKey.lastIndexOf("/") + 1);
+		final String userJarFileURI = "/tmp/" + userJarKey.substring(userJarKey.lastIndexOf("/") + 1);
+
+		LOG.info("s3SinkPath is :" + s3SinkPath);
+		LOG.info("region is :" + region);
+		LOG.info("inputStreamName is :" + inputStreamName);
+		LOG.info("userJarBucket is :" + userJarBucket);
+		LOG.info("userJarKey is :" + userJarKey);
+
+		downloadUserJars(region, userJarKey, userJarFileURI,userJarBucket);
+
+		LOG.info("Starting the application...");
+		final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+
+		// If the referenced JAR is needed to run as part of the Flink operators, usually it's a best practice to
+		// ... download the jar files and save them in the cache store. This way Task Managers can access them with no delay.
+		env.registerCachedFile(userJarFileURI, userJarKey);
+		env.getCheckpointConfig().setCheckpointInterval(Time.minutes(1).toMilliseconds());
+
+		DataStream<String> input = createSourceFromStaticConfig(env, region, inputStreamName);
+		SingleOutputStreamOperator<Tuple2<String, Integer>> tokenizedStream = input
+				.process(new ProcessTokenizer(userJarKey));
+
+		DataStream<String> inputStream = tokenizedStream
+				.keyBy(event -> event.f0)
+				.window(TumblingEventTimeWindows.of(Time.seconds(15)))
+				.sum(1)
+				.map(value -> value.f0 + " Count: " + value.f1 + "\n");
+
+		inputStream.addSink(createS3SinkFromStaticConfig(s3SinkPath));
+		env.execute("Flink Streaming Java API Skeleton");
+	}
+
+	private static void downloadUserJars(String region, String userJarKey,String userJarFileURI, String userJarBucket) throws IOException {
+		AmazonS3 s3Client = AmazonS3ClientBuilder.standard()
+				.withRegion(region)
+				.withCredentials(new DefaultAWSCredentialsProviderChain())
+				.build();
+		// Get an object and print its contents.
+		LOG.info("Downloading " + userJarKey);
+
+		S3Object fileObject = s3Client.getObject(new GetObjectRequest(userJarBucket, userJarKey));
```

Comment:
This seems rather complex. Any reason you can't use `getObject(getObjectRequest, destinationPath)` instead?

Reply:
There is no such overload in my s3Client.

Reply:
This is part of the v2 sdk: https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/S3Client.html.

Reply:
Got it! changed and replaced the code.